File: DSA_Level-1/Basics Of Programming/2D Arrays/Spiral Display/SpiralDisplay_Solution.js

function print(nums){
    let minRow=0,maxRow=nums.length-1;
    let minCol=0,maxCol=nums[0].length-1;

    let count=0, printCount=nums.length*nums[0].length;

    while(count<printCount){
        
        if(count<printCount){
            for(let i=minRow;i<=maxRow;++i){
                console.log(nums[i][minCol]);
                ++count;
            }
        }

        minCol++;

        if(count<printCount){
            for(let i=minCol;i<=maxCol;++i){
                console.log(nums[maxRow][i]);
                ++count;
            }
        }

        maxRow--;

        if(count<printCount){
            for(let i=maxRow;i>=minRow;--i){
                console.log(nums[i][maxCol]);
                ++count;
            }
    
        }

        maxCol--;

        if(count<printCount){
            for(let i=maxCol;i>=minCol;--i){
                console.log(nums[minRow][i]);
                ++count;
            }
        }

        minRow++;

    }
}

print([[1,2,3],[1,3,4],[1,2,5]]);

File: DSA_Level-1/Basics Of Programming/2D Arrays/Spiral Display/SpiralDisplay_Solution.java

import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws Exception {
        Scanner scn=new Scanner(System.in);
        int n=scn.nextInt();
        int m=scn.nextInt();
        
        int nums[][]=new int[n][m];
        
        for(int i=0;i<n;++i){
            for(int j=0;j<m;++j){
                nums[i][j]=scn.nextInt();
            }
        }
        
        int count=0, printCount=n*m;
        
        int minRow=0;
        int maxRow=n-1;
        int minCol=0;
        int maxCol=m-1;
        
        while(count<printCount){
            if(count<printCount){
                for(int i=minRow;i<=maxRow;++i){
                    System.out.println(nums[i][minCol]);
                    ++count;
                }
                minCol++;
            }
            
            if(count<printCount){
                for(int i=minCol;i<=maxCol;++i){
                    System.out.println(nums[maxRow][i]);
                    ++count;
                }
                maxRow--;
            }
            
            if(count<printCount){
                for(int i=maxRow;i>=minRow;--i){
                     System.out.println(nums[i][maxCol]);
                    ++count;
                }
                --maxCol;
            }
            
            if(count<printCount){
                for(int i=maxCol;i>=minCol;--i){
                    System.out.println(nums[minRow][i]);
                    ++count;                    
                }
                minRow++;
            }
        }
    }

}